### DISCUSSION TITLE
async connection pool

### DISCUSSION THREAD

**Discussion by jigneshmpatel** [redis/lettuce#1593] [Q&A] [answered]
I am exploring concept of using in Async Connection pool and wondering is that right approach, since lettuce inherantly supports pipelining?

**Comment by mp911de** [comment, marked as answer]
Actually, pooling is only required when using transactions or blocking operations (`BLPOP`, `XREAD … BLOCK …`). In all other cases, you're better off with a single connection.

**Comment by jigneshmpatel** [comment]
Thanks Mark for immediate response!!
In that case, I do better off using sync connection pooling for the blocking operations.

**Reply by mp911de** [reply]
Note that blocking redis operations are a different kind of blocking than Java blocking (synchronous) calls.

**Comment by jigneshmpatel** [comment]
Mark,
Please elaborate more about your last statement.

Moreover I tried the example to create connection pool using cluster as per the documentation
[https://lettuce.io/core/release/reference/#connection-pooling.blocking](url)
[https://github.com/lettuce-io/lettuce-core/wiki/Connection-Pooling](url)
```
GenericObjectPool<StatefulRedisClusterConnection<String, String>> pool = ConnectionPoolSupport
               .createGenericObjectPool(() -> clusterClient.connect(), new GenericObjectPoolConfig());
```
but it gives following error.
<img width="951" alt="Screen Shot 2021-01-19 at 10 44 55 PM" src="https://user-images.githubusercontent.com/780048/105124473-33808b80-5aa8-11eb-8d3e-4b76bf1ff26a.png">
